### PR TITLE
feat(cdk/portal): allow specifying injector for template portal

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

**Review by crisbeto — APPROVED** [review]
LGTM

File: src/cdk/portal/portal-directives.ts [hunk 1]
```diff
@@ -181,7 +181,9 @@ export class CdkPortalOutlet extends BasePortalOutlet implements OnInit, OnDestr
    */
   attachTemplatePortal<C>(portal: TemplatePortal<C>): EmbeddedViewRef<C> {
     portal.setAttachedHost(this);
-    const viewRef = this._viewContainerRef.createEmbeddedView(portal.templateRef, portal.context);
+    const viewRef = this._viewContainerRef.createEmbeddedView(portal.templateRef, portal.context, {
+      injector: portal.injector,
+    });
```
Comment (lekhmanrus) [inline src/cdk/portal/portal-directives.ts:186]: Should it be
```ts
    const viewRef = this._viewContainerRef.createEmbeddedView(
      portal.templateRef,
      portal.context, portal.injector ? { injector: portal.injector } : undefined
    );
```
instead?

See [StackBlitz](https://stackblitz.com/edit/angular-mnrvoq?file=package.json). When we click **Render template portal** button it throws an error:
image

Comment (crisbeto) [reply src/cdk/portal/portal-directives.ts:186]: I can't reproduce this error locally in our dev app. I think the problem is that your Stackblitz is running v14 of the CDK against v13 of the Framework which doesn't support the `injector` option yet.

Comment (lekhmanrus) [reply src/cdk/portal/portal-directives.ts:186]: @crisbeto, right. Sorry about that.

**Pull request merged**